ratis-examples/src/main/java/org/apache/ratis/examples/filestore/FileStore.java

```diff
@@ -219,4 +225,67 @@ public void close() {
     reader.shutdownNow();
     deleter.shutdownNow();
   }
+
+  CompletableFuture<StreamWriteReplyProto> streamCommit(String p, long bytesWritten) {
+    return CompletableFuture.supplyAsync(() -> {
+      final Path full = buildFullPath(p);
+      try {
+        RandomAccessFile file = new RandomAccessFile(full.toFile(), "r");
+        long len = file.length();
+        return StreamWriteReplyProto.newBuilder().setIsSuccess(len == bytesWritten).setByteWritten(len).build();
+      } catch (FileNotFoundException e) {
+        throw new CompletionException("Failed to find " + p, e);
+      } catch (IOException e) {
+        throw new CompletionException("Failed to get length of " + p, e);
```

Comment:
If IOException happens, we can not make sure the reason is "Failed to get length of "

Reply:
It will:

`RandomAccessFile file = new RandomAccessFile(full.toFile(), "r");` will throw FileNotFoundException, thus match the first exception.
If the file indeed exists, `file.length();` will give IOException if anything wrong, thus hit the second exception.

And exception that is matched first gets executed: https://stackoverflow.com/questions/10964882/order-of-catching-exceptions-in-java#:~:text=The%20order%20is%20whatever%20matches,is%20matched%20or%20none%20are.

Reply:
In fact, because now I use `Path resolve(Path relative) throws IOException` so the order of exception is no longer holds. I have merged all into catching IOException